### PR TITLE
Fixed #21734 -- Handled ProtectedError on POST to admin's delete_selected action

### PR DESCRIPTION
**Pull request by akki**
https://code.djangoproject.com/ticket/21734

Approach is similar to that used to solve [#26235](https://code.djangoproject.com/ticket/26235) in https://github.com/django/django/commit/49ac10b4de03279da9d67aeac5cb8561fc055085.

### CODE REVIEW

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -2070,10 +2070,20 @@ def test_post_delete_protected(self):
         q = Question.objects.create(question='Why?')
         Answer.objects.create(question=q, answer='Because.')
 
+        # If the protected object is tried to be deleted using the delete view.
         response = self.client.post(reverse('admin:admin_views_question_delete', args=(q.pk,)), {'post': 'yes'})
         self.assertEqual(Question.objects.count(), 1)
         self.assertContains(response, "would require deleting the following protected related objects")
 
+        # If the protected object is tried to be deleted using the delete_selected action.
```
Comment (timgraham) [inline tests/admin_views/tests.py]: Please put the test in `AdminActionsTest`.

Comment (akki) [inline tests/admin_views/tests.py]: Oops. Done.

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -3156,6 +3156,22 @@ def test_model_admin_default_delete_action_no_change_url(self):
...
 
+    def test_post_delete_action_protected(self):
+        """
+        A POST request to delete protected objects should display the page
+        which says the deletion is prohibited.
+        """
+        q = Question.objects.create(question='Why?')
+        Answer.objects.create(question=q, answer='Because.')
+
+        response = self.client.post(reverse('admin:admin_views_question_changelist'), {
```
Comment (timgraham) [inline tests/admin_views/tests.py]: I think you could add this to `test_model_admin_default_delete_action_protected` so you don't have to repeat the setup steps.

**Comment by akki** [comment]
Done.

**Comment by timgraham** [comment]
merged in a7c813ba045044ec23bb656ef9a23a0e38e36514, thanks!

**Pull request closed without merging**